src/Action.hs

```diff
@@ -144,14 +191,6 @@ instance GenArbitrary r => GenArbitrary (Action r) where
 
 -- * concrete monad type; user state
```

Comment:
The `UserState` type is defined much higher up. And if the comment means the monad's main purpose is to hold user state, then the wording confused me.